test/adapter/json_api/relationship_test.rb

```diff
@@ -19,19 +19,25 @@ class RelationshipAuthorSerializer < ActiveModel::Serializer
 
             has_many :locations do
               link :related do
-                ids = object.locations.map!(&:id).join(',')
+                ids = object.locations.map(&:id).join(',')
```

Comment:
I was using `map!` instead of `map` with actually change the array and might cause some issues.